app/models/move.rb

```diff
@@ -157,7 +164,7 @@ def set_reference
   end
 
   def set_move_type
```

Comment:
We could move this logic to the move_actions v1? this could help avoid having to define anything regarding v2

Reply:
Explored this approach and it didn't end well. The problem is we have many (200+) existing tests that currently use the move factory and then go on to set invalid combinations of move and location types. I was able to resolve most of those without making the PR too noisy by setting the default move_type in the factory to `nil`. That then means that `set_move_type` is called when the model is validated and move type is correctly determined based on the location types set for each test case. Removing the `before_validation` callback on this method and calling it explicitly from the `create_and_render` method in v1 works for the create move request specs, but reintroduces the issue with inconsistent test data in other specs and exposes the 200+ broken specs.

I think it's just too noisy to fix all of the broken specs as part of this PR so my preference is to keep the implementation as is for now (it's temporary anyway), and address the broken specs at a later stage; when we remove support for v1 there will be a lot of redundant code and so many of these broken specs will also be removed and there will be a lot less stuff to fix.